modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/TransportRankEvalAction.java

```diff
@@ -75,46 +75,36 @@ public TransportRankEvalAction(Settings settings, ThreadPool threadPool, ActionF
-        RankEvalResponse response = new RankEvalResponse();
         RankEvalSpec qualityTask = request.getRankEvalSpec();
         RankedListQualityMetric metric = qualityTask.getEvaluator();
 
-        for (QuerySpec spec : qualityTask.getSpecifications()) {
-            double qualitySum = 0;
-
+        double qualitySum = 0;
+        Map<String, Collection<String>> unknownDocs = new HashMap<String, Collection<String>>();
+        Collection<QuerySpec> specifications = qualityTask.getSpecifications();
+        for (QuerySpec spec : specifications) {
             SearchSourceBuilder specRequest = spec.getTestRequest();
-            String[] indices = new String[spec.getIndices().size()]; 
+            String[] indices = new String[spec.getIndices().size()];
             spec.getIndices().toArray(indices);
             SearchRequest templatedRequest = new SearchRequest(indices, specRequest);
 
+            TransportSearchAction transportSearchAction = new TransportSearchAction(settings, threadPool, searchPhaseController,
+                    transportService, searchTransportService, clusterService, actionFilters, indexNameExpressionResolver);
+            ActionFuture<SearchResponse> searchResponse = transportSearchAction.execute(templatedRequest);
+            SearchHits hits = searchResponse.actionGet().getHits();
 
-            Map<Integer, Collection<String>> unknownDocs = new HashMap<Integer, Collection<String>>();
-            Collection<RatedQuery> intents = qualityTask.getIntents();
-            for (RatedQuery intent : intents) {
-
-                TransportSearchAction transportSearchAction = new TransportSearchAction(
-                        settings, 
-                        threadPool, 
-                        searchPhaseController, 
-                        transportService, 
-                        searchTransportService, 
-                        clusterService, 
-                        actionFilters, 
-                        indexNameExpressionResolver);
-                ActionFuture<SearchResponse> searchResponse = transportSearchAction.execute(templatedRequest);
-                SearchHits hits = searchResponse.actionGet().getHits();
-
-                EvalQueryQuality intentQuality = metric.evaluate(hits.getHits(), intent);
-                qualitySum += intentQuality.getQualityLevel();
-                unknownDocs.put(intent.getIntentId(), intentQuality.getUnknownDocs());
-            }
-            response.addRankEvalResult(spec.getSpecId(), qualitySum / intents.size(), unknownDocs);
+            EvalQueryQuality intentQuality = metric.evaluate(hits.getHits(), spec.getRatedDocs());
+            qualitySum += intentQuality.getQualityLevel();
+            unknownDocs.put(spec.getSpecId(), intentQuality.getUnknownDocs());
         }
+        RankEvalResponse response = new RankEvalResponse();
+        RankEvalResult result = new RankEvalResult(qualityTask.getTaskId(), qualitySum / specifications.size(), unknownDocs);
```

Comment:
+1 Also relates to my comment above about which average to provide. Come to think of it a bit more, it might even make sense to provide summarisation statistics other than average.